PowerShellGet/PSModule.psm1

```diff
@@ -8584,22 +8596,53 @@ function ValidateAndAdd-PSScriptInfoEntry
                             $KeyName = $script:Version
 
                             [Version]$Version = $null
-
-                            if([System.Version]::TryParse($Value, ([ref]$Version)))
+                            
+                            # Special case for prerelease string in version field, must pull out before parsing version
+                            if ($Value -match '-') 
                             {
-                                $Value = $Version                            
+                                [String]$PreRelease = $null
+
+                                $indexOfFirstHyphen = $Value.IndexOf('-')

```

Comment:
I saw this pattern repeated..is it possible to capture this in a function and reuse?